sdk/core/src/pipeline.rs

```diff
@@ -0,0 +1,96 @@
+use crate::policies::{Policy, PolicyResult, RetryPolicy};
+use crate::{Context, Request, Response};
+use std::sync::Arc;
+
+#[derive(Clone)]
+pub struct Pipeline {
+    per_call_policies: Vec<Arc<dyn Policy<Request>>>,
+    retry: Arc<dyn RetryPolicy>,
+    per_retry_policies: Vec<Arc<dyn Policy<Request>>>,
+    transport: Arc<dyn Policy<Response>>,
+}
+
+impl std::fmt::Debug for Pipeline {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "Pipeline{{per_call_policies.len() == {}, per_retry_policies.len() == {}}}",
```

Comment:
Not sure how helpful this debug info is. We can leave it for now though.

Reply:
It's really not that useful. But I do not see how to make it more informative short of constraining `Policy` to be `Debug` as well 🤷‍♂️